tock/hours/views.py

```diff
@@ -613,6 +616,71 @@ def get_object(self, queryset=None):
             user_id=self.request.user.id)
         return obj
 
+    def clean_task_data(self, float_people_id):
+        # Get Float task data for week corresponding w/ ReportingPeriod
+        # instance.
+        r = get_float_data(
+            endpoint='tasks',
+            params={
+                'weeks': base.FLOAT_API_TASK_WEEKS,
+                'start_day': datetime.datetime.strptime(
+                    self.kwargs['reporting_period'], "%Y-%m-%d"
+                ).date()
+            }
+        )
+        if r.status_code != 200:
+            return None
+        else:
+            task_data = r.json()
+        #From response, derive meta data and tasks for the period.
+        f_start_date = datetime.date(
+            int(task_data['start_yr']), 1, 1) + datetime.timedelta(
+            days=int(task_data['start_doy']
+            )-1)
+        f_end_date = f_start_date + datetime.timedelta(
+            days=(base.FLOAT_API_TASK_WEEKS * 6))
+        clean_data = {
+            'metadata': {
+                'float_period_start': f_start_date,
+                'float_period_end': f_end_date
+            },
+            'tasks': [ item for sublist in \
+                [ ii for ii in [i['tasks'] for i in task_data['people'] \
+                if i['people_id'] == float_people_id] ] \
+            for item in sublist ]
+        }
+        # Calculate estimated hours per week. Need to take into account
+        # holidays in future build.
```

Comment:
Would definitely work for holidays (assuming there isn't any Federal employee weirdness around administrative closings that might not be captured by `businesstime`, but also need to figure out individual time off (see the [/timeoffs](https://github.com/floatschedule/api/blob/master/Sections/timeoff.md) endpoint). Dove into this over on #635.